server/application/application.proto

```diff
@@ -54,6 +54,7 @@ message ApplicationResponse {}
 message ApplicationCreateRequest {
 	required github.com.argoproj.argo_cd.pkg.apis.application.v1alpha1.Application application = 1 [(gogoproto.nullable) = false];
 	optional bool upsert = 2;
+	optional bool validate = 3;
```

Comment:
For consistency please add `validate` field to `ApplicationUpdateRequest` as well.